### PR TITLE
Instant Search: add image alt text from API

### LINKED ISSUE #19273: Instant Search: use correct image alt text
#### Is your feature request related to a problem? Please describe.
When uploading an image in WordPress' media library, it is possible to specify the alt text for it. We currently use the post title as alt text in search results, but it would be better to use the alt text the user has specified wherever it is available.

Image-attachment-details

#### Describe the solution you'd like
The ElasticSearch API returns `image.alt` in the response and we use that for image alt text in the application.

cc @gibrown

### PR DESCRIPTION
**Pull request by bluefuton**
Fixes #19273.

#### Changes proposed in this Pull Request:
The Instant Search API now provides `image.alt_text` as a field we can request.

This PR adds the new field and switches to use it for alt text in the expanded and product result formats.

I've also reduced the fields fetched for the minimal result format (image fields are not needed).

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions
1. On a site with Jetpack Instant Search enabled, create a post with an image. 
2. Set alt text for the image and ensure that the alt text appears on images in search results too (`/?s=`).

Screen Shot 2022-01-11 at 15 57 40

### CODE REVIEW

File: projects/packages/search/src/instant-search/components/search-result-expanded.jsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default function SearchResultExpanded( props ) {
 					<div className="jetpack-instant-search__search-result-expanded__image-container">
 						{ firstImage ? (
 							<PhotonImage
-								alt={ fields[ 'title.default' ] }
+								alt={ fields[ 'image.alt_text' ] }
```
Comment (bluefuton) [inline projects/packages/search/src/instant-search/components/search-result-expanded.jsx:80]: I thought about keeping `title.default` as a fallback here, but it's complicated by the fact that `alt=""` is [valid alt text](https://www.w3.org/WAI/tutorials/images/decorative/). 

Feels best to rely on the alt text from the API in all cases.

**Review by kangzj — COMMENTED** [review]
This tests well and looks good to me 👍 
Some checks are failing, and the errors don't seem relevant to the PR - probably just need to rebase and have the checks re-run and see.

**Comment by bluefuton** [comment]
The only required check failing is 'Linting / Changelogger validity (pull_request)':

Screen Shot 2022-01-13 at 14 53 15

@Automattic/jetpack-crew can you advise whether we need to use fixup-project-versions here?

**Comment by anomiex** [comment]
> @Automattic/jetpack-crew can you advise whether we need to use fixup-project-versions here?

Well, you don't _need_ to, you could do the version bumps manually. 😀 But fixup-project-versions saves the trouble of figuring out all the details.

P.S. if you're on a Mac and that command is really slow, you might see p7H4VZ-3gs-p2 (or https://apple.stackexchange.com/questions/338415/php-incredibly-slow-since-upgrading-macos-to-10-14-mojave)

**Comment by anomiex** [comment]
> Thanks @anomiex! Do we need to do this on every search PR in the future?

I usually run it on my PRs that touch any project, just before pushing. It doesn't take that long and it'll safely do nothing if nothing is needed.

In general for a composer package it's only needed if you're doing the first "minor" or "major" PR since the last time the package was released (which will generally be weekly, since packages used by Jetpack get released as part of the weekly Jetpack release for Atomic), unless the package is configured with a version number constant.

File: projects/plugins/jetpack/changelog/add-instant-search-alt-text-where-available#2 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
```
Comment (kangzj) [inline projects/plugins/jetpack/changelog/add-instant-search-alt-text-where-available#2:1]: Do you think keeping `projects/plugins/jetpack/changelog/add-instant-search-alt-text-where-available` is sufficient enough?

**Review by kangzj — DISMISSED** [review]
Good job! It looks good and works well 👍

**Review by jeherve — CHANGES_REQUESTED** [review]
I'm afraid this is going to need a rebase now. :(

**Review by jeherve — APPROVED** [review]

**Pull request merged**